components/identifier-generator/back/src/Domain/Model/Property/FamilyProperty.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Akeneo\Pim\Automation\IdentifierGenerator\Domain\Model\Property;
+
+use Webmozart\Assert\Assert;
+
+/**
+ * @copyright 2023 Akeneo SAS (https://www.akeneo.com)
+ * @license   https://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ *
+ * @phpstan-type FamilyPropertyNormalized array{type: 'family', process: ProcessNormalized}
+ */
+final class FamilyProperty implements PropertyInterface
+{
+    public function __construct(
```

Comment:
I don't find any new FamilyProperty(...), please set this private
```suggestion
    private function __construct(
```